### ISSUE TITLE
Cleaning up the open source neutronics plugin

### ISSUE THREAD

**Issue by albeanth** [terrapower/armi#1452]
https://github.com/terrapower/armi/blob/753660d71f3777680d541ff5d042d2d71100f4a3/armi/physics/neutronics/__init__.py#L29-L31

The creation of this issue stems from the above statement. In the hope of addressing some overdue technical debt, I just wanted to create this Issue so we could try and tackle this piecemeal. 

A few ideas:
- [ ] Split `NeutronicsPlugin` into a `plugin.py` file within the armi/physics/neutronics directory. This follows the set up of other plugins more closely. 
- [ ] move the constants defined here into a `constants.py` file
- [ ] clean out any unused constants... This is debatable, but I don't think ARMI should be responsible for owning constants used in downstream projects. This is a whole can-o-worms... I know. 
- [ ] move "settings checks" closer to the interfaces they're used in OR to settings validators in the `settings.py` file. 
- [ ] Can the LatticePhysicsFrequency numeration get moved to settings.py? might be a bad idea, haven't thought critically about it.

**Comment by john-science** [comment]
Some of these really don't need to be here:

https://github.com/terrapower/armi/blob/1636d23787671a91e240eaae24740a7e2e7e979e/armi/physics/neutronics/__init__.py#L175-L177

https://github.com/terrapower/armi/blob/1636d23787671a91e240eaae24740a7e2e7e979e/armi/physics/neutronics/__init__.py#L160

https://github.com/terrapower/armi/blob/1636d23787671a91e240eaae24740a7e2e7e979e/armi/physics/neutronics/__init__.py#L147

This is not time well spent. And most of these are only used in one project downstream.

**Comment by drewj-tp** [comment]
A reflective boundary condition is a common, broadly applicable boundary condition for nuclear engineering applications

**Comment by john-science** [comment]
> A reflective boundary condition is a common, broadly applicable boundary condition for nuclear engineering applications

Yeah, but this is just the _string_ "Reflective". It's not like that _string_ is used in ARMI. It's used in one repo downstream.

Maybe someone else would want that string in their code, sure. But maybe they would also want it all lower case. And... should ARMI be in the business of defining every single word and term that someone in the nuclear industry might want, as global constants?  I am unconvinced this is notably helpful as a feature in ARMI.

**Comment by john-science** [comment]
@yardasol Three of the above 5 bullet points are still open. See if you think any of these make sense to you. Basically, Tony's idea here is to make the neutronics plugin in ARMI look like the other plugins in ARMI: to help organize the code. On balance, I think you could do any of these three tasks. But I think the interesting part is starting to look at how ARMI plugins are defined and laid out.

Your call. Just throwing it out there.
